wgpu-core/src/command/ray_tracing.rs

```diff
@@ -173,7 +173,8 @@ impl Global {
         let mut scratch_buffer_blas_size = 0;
         let mut blas_storage = Vec::new();
         let mut cmd_buf_data = cmd_buf.data.lock();
-        let cmd_buf_data = cmd_buf_data.as_mut().unwrap();
+        let mut cmd_buf_data_guard = cmd_buf_data.record()?;
+        let cmd_buf_data = &mut *cmd_buf_data_guard;
```

Comment:
There's a `return Ok(...)` after computing `scratch_size`, below. Does that need a `mark_successful` call too?